Observer/Event/SendAddToWishlistEvent.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tweakwise\Magento2Tweakwise\Observer\Event;
+
+use Exception;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Checkout\Model\Session;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+use Psr\Log\LoggerInterface;
+use Tweakwise\Magento2Tweakwise\Model\Client;
+use Tweakwise\Magento2Tweakwise\Model\Client\RequestFactory;
+use Tweakwise\Magento2Tweakwise\Model\PersonalMerchandisingConfig;
+use Tweakwise\Magento2TweakwiseExport\Model\Helper;
+
+class SendAddToWishlistEvent implements ObserverInterface
+{
+    /**
+     * @param PersonalMerchandisingConfig $config
+     * @param RequestFactory $requestFactory
+     * @param Client $client
+     * @param LoggerInterface $logger
+     * @param Helper $helper
+     * @param StoreManagerInterface $storeManager
+     * @param Session $checkoutSession
+     */
+    public function __construct(
+        private readonly PersonalMerchandisingConfig $config,
+        private readonly RequestFactory $requestFactory,
+        private readonly Client $client,
+        private readonly LoggerInterface $logger,
+        private readonly Helper $helper,
+        private readonly StoreManagerInterface $storeManager,
+        private readonly Session $checkoutSession,
+    ) {
+    }
+
+    /**
+     * @param Observer $observer
+     * @return void
+     */
+    public function execute(Observer $observer): void
+    {
+        try {
+            if (!$this->config->isAnalyticsEnabled()) {
+                return;
+            }
+
+            $this->sendAddToWishlistEvent($observer->getEvent()->getProduct());
+        } catch (Exception $e) {
+            $this->logger->error('Tweakwise Add To Wishlist event could not be sent', ['message' => $e->getMessage()]);
+            return;
+        }
+    }
+
+    /**
+     * @param ProductInterface $product
+     * @return void
+     * @throws NoSuchEntityException
+     */
+    protected function sendAddToWishlistEvent(ProductInterface $product): void
+    {
+        if (!$product instanceof Product) {
+            return;
+        }
+
+        $tweakwiseRequest = $this->requestFactory->create();
+        $tweakwiseRequest->setParameter('ProfileKey', $this->config->getProfileKey());
+        $tweakwiseRequest->setParameter('SessionKey', $this->checkoutSession->getSessionId());
```

Comment:
@jansentjeu I think this sessionKey needs to be the same sessionKey you use with the session start event. If it is the same, you can ignore this comment.